src/lib.rs

```diff
@@ -51,15 +51,55 @@ impl Pool {
         pool
     }
 
+    /// Create a new Pool with `size` threads and given thread config.
+    ///
+    /// If `size` is zero, no threads will be spawned. Threads can
+    /// be added later via `expand`.
+    ///
+    /// NOTE: Since Pool can be freely cloned, it does not represent a unique
+    /// handle to the thread pool. As a consequence, the thread pool is not
+    /// automatically shut down; you must explicitly call `Pool::shutdown` to
+    /// shut down the pool.
+    #[inline]
+    pub fn new_with_thread_config(size: usize, thread_config: ThreadConfig) -> Pool {
```

Comment:
Convention is usually to just call constructors like these `with_thread_config`.